src/@types/index.ts

```diff
@@ -0,0 +1,415 @@
+import * as Beacon from "../lib/beacon"
+import { KnownErrors } from "../lib/errors"
+
+/**
+ * An interface representing simple objects mapping strings to basic primative
+ * types.
+ */
+export interface SimpleObject {
+    [key: string]:
+        | SimpleObject
+        | SimpleObject[]
+        | Array<string | number | boolean | null>
+        | boolean
+        | number
+        | string
+        | null
+        | undefined
+}
+
+/**
+ * An interface for objects containing a set of executable tasks.
+ */
+export interface ExecutableContainer {
+    /**
+     * A list of {@link Executable} objects.
+     */
+    executables: Executable[]
+}
+
+/**
+ * A type for objects returned by {@link Provider.fetchSessionConfig}
+ * representing configuration for a RUM session.
+ */
+export type SessionConfig = ExecutableContainer
+
+/**
+ * An interface representing the configuration of a RUM test.
+ */
+export interface TestConfiguration {
+    /**
+     * Indicates the test type.
+     */
+    type: string
+}
+
+/**
+ * An interface representing the result of a test's setup activity.
+ */
+export interface TestSetupResult {
+    /**
+     * A container for the test setup data
+     */
+    data?: SimpleObject
+}
+
+/**
+ * A type representing a Resource Timing API entry having _entryType_
+ * "resource".
+ *
+ * Alias for {@link SimpleObject}
+ */
+export type ResourceTimingEntry = SimpleObject
```

Comment:
Note: I've battled with this myself in `insights.js` and `performance-observer-polyfill`. Ideally, I'd prefer to use an offical type exported by the TypeScript DOM library, such as `PerformanceEntry` but it's never been sufficient for me and the `toJSON()` method always gets in the way.

From the TypeScript source: https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L11506-L11512  
```
interface PerformanceEntry {
    readonly duration: number;
    readonly entryType: string;
    readonly name: string;
    readonly startTime: number;
    toJSON(): any;
}
``` 

Maybe we could try using one of the fancy utility types, such as [`Omit<T, K>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk) here? Such as:
```suggestion
export type ResourceTimingEntry = Omit<PerformanceEntry, 'toJSON'>
```
I haven't tried this so don't know if it will work. 🤣